docs/source/federated-types/composition.mdx

```diff
@@ -248,3 +248,254 @@ type User @key(fields: "id") {
 If you make a shared type an [entity](../entities/), different subgraphs can define any number of different fields for that type, as long as they all define **key fields** for it.
 
 This is a useful solution when a type corresponds closely to an entry in a data store that one or more of your subgraphs has access to (e.g., a `Users` database).
+
+## Merging types from multiple subgraphs
+
+If a particular GraphQL type is defined _differently_ by different subgraphs, composition uses one of two strategies to merge those definitions: **union** or **intersection**.
+
+* **Union**: The supergraph schema includes _all_ parts of _all_ subgraph definitions for the type. 
```

Comment:
The downside of structuring it this way is that you never really have the opportunity to talk about the "why" of the reason each strategy is employed. In the case of input object types, we use intersection because it allows us to guarantee that no subgraph ever has to deal with input that it didn't expect. In the case of output types, it's safe to compose all the different options together because you don't have the restriction the same way you do with input validation.

Reply:
So, we do have one sentence in the "Input types and field arguments" section mentioning the logic behind the intersection strategy:

> Composition _always_ uses the intersection strategy to merge input types and field arguments. This ensures that the gateway _never_ passes an argument to a subgraph that doesn't define that argument.

Would you like me to add some more emphasis on this point, @clenfest?

Reply:
If not, @StephenBarlow can you merge this? (I don't have write access) 

I can also always add more context to this later if folks need it! :)

Reply:
@rkoron007 Build is currently failing for unrelated reasons 🤔 I will explore in a bit and we can get this merged!

Reply:
Running `npm run codegen` and checking in the subsequent changed files should fix the build. I don't mind doing that, let me know (didn't want to step on toes if you're already on it).

Reply:
Oh! I think I've done that previously. Might be best to merge the latest from `main`.

Reply:
Thank you @benweatherman! Did as you suggested, and it looks like everything should be good to go now @StephenBarlow ✨